### ISSUE TITLE
AudioPlayer.PlaybackFailed not handled

### ISSUE THREAD

**Issue by colinfrei** [dmhacker/alexa-youtube-skill#2]
When I try to run this, I get an error when trying to play back the media file. This is sent as a 'AudioPlayer.PlaybackFailed' request, which doesn't have the applicationid in the sessiondetails, and that causes the following error:
>Unhandled exception: Cannot read property 'applicationId' of undefined.

I think the 'pre' function needs to handle that case as well, and the app should probably log errors if it receives them.

**Comment by angelos1989** [comment]
@colinfrei did you manage to fix this? I have the same issue

**Comment by dmhacker** [comment]
I updated the alexa-app package so request.applicationId returns the correct information in app.pre. See if that works.

**Comment by kadeschs** [comment]
It seems to find stuff.  I had a few successful finds.  However, it doesn't play anything afterward.

**Comment by Firni95** [comment]
Same here. Got the answer: {"state":"success","message":"Attempting upload ...","link":"https://s3-us-west-1.amazonaws.com/alexa-youtube-skill/audio/UC81i2M30Bc.mp3"} but it doesn't play any music

**Comment by dmhacker** [comment]
This is a known issue that Alexa doesn't play the audio. Can you guys @kadeschs @Firni95 @angelos1989 test this by giving the skill something to look up and then looking through your Lambda's CloudWatch logs? You might be getting something like:

`{ message: 'Your GStreamer installation is missing a plug-in.',
type: 'MEDIA_ERROR_UNKNOWN' }`

**Comment by kadeschs** [comment]
Yes. It says it's missing the plug-in.

**Comment by angelos1989** [comment]
@dmhacker I have that issue with the Gstreamer as you described. Can we change the content type as suggested by @kadeschs and here : https://forums.developer.amazon.com/questions/59814/audioplayer-live-stream.html

**Comment by dmhacker** [comment]
@kadeschs  @angelos1989 I'm going to update [the server](https://github.com/dmhacker/dmhacker-youtube) to return a m3u file (content-type would then equal application/x-mpegurl). The m3u file basically would serve as a redirect to the actual mp3 file, which will still be hosted on the same S3 bucket.

**Comment by ohadbenita** [comment]
Hi, I'm having the same issue as described with the missing plugin, see my latest Lambda invocation -

START RequestId: 5c7d0039-76ea-11e7-bdd0-534cefdb0c5f Version: $LATEST
2017-08-01T18:51:01.208Z	5c7d0039-76ea-11e7-bdd0-534cefdb0c5f	Searching ... this besito
2017-08-01T18:51:01.537Z	5c7d0039-76ea-11e7-bdd0-534cefdb0c5f	Found ... Luis Fonsi, Daddy Yankee - Despacito (letra official) ft. Justin Bieber - lyrics official.
2017-08-01T18:51:02.320Z	5c7d0039-76ea-11e7-bdd0-534cefdb0c5f	Processed.
2017-08-01T18:51:02.379Z	5c7d0039-76ea-11e7-bdd0-534cefdb0c5f	Stored @ https://s3-us-west-1.amazonaws.com/alexa-youtube-skill/audio/zlabsbE4aiQ.m3u8
END RequestId: 5c7d0039-76ea-11e7-bdd0-534cefdb0c5f
REPORT RequestId: 5c7d0039-76ea-11e7-bdd0-534cefdb0c5f	Duration: 1179.53 ms	Billed Duration: 1200 ms Memory Size: 384 MB	Max Memory Used: 40 MB	

Any update on how to fix this issue?
@dmhacker Your help is highly appreciated

**Comment by kadeschs** [comment]
@dmhacker Try looking at https://github.com/Fenton-Fenton-Fenton/alexa-tube. They are accomplishing something similar with success.  They are passing the mp3 file temporarily to a dropbox account.  Might shed some light on what's going on here.

**Comment by dmhacker** [comment]
@ohadbenita @kadeschs @Firni95 @angelos1989 

This is now fixed! The issue was that the mp3 file downloaded with ytdl-core in the Heroku server wasn't the correct bitrate. I fixed this by using ffmpeg to resave the file and modify the bitrate. 

Yippee! Finally, no more GStreamer issues. If you are getting an issue or something isn't working, make sure everything you've done on your end matches the setup instructions in the README.
